### PR TITLE
[API] Browsing promotions

### PR DESCRIPTION
**Pull request by arti0090**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Api/Admin/ManagingPromotionsContext.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Behat\Client\ResponseCheckerInterface;
+use Sylius\Component\Core\Model\PromotionInterface;
+use Webmozart\Assert\Assert;
+
+final class ManagingPromotionsContext implements Context
+{
+    /** @var ApiClientInterface */
+    private $apiAdminClient;
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingPromotionsContext.php:25]: As we are in the admin context:
```suggestion
    private $client;
```
?

File: src/Sylius/Behat/Context/Api/Admin/ManagingPromotionsContext.php [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+     * @Then the :promotionName promotion should exist in the registry
+     */
+    public function thePromotionShouldAppearInTheRegistry(string $promotionName): void
+    {
+        $promotion = $this->getPromotionByPromotionNameFromResponseCollection(
+            $promotionName,
+            $this->responseChecker->getCollection($this->apiAdminClient->getLastResponse())
+        );
+
+        Assert::notNull($promotion);
```
Comment (AdamKasp) [inline src/Sylius/Behat/Context/Api/Admin/ManagingPromotionsContext.php:69]:
```suggestion
        Assert::notNull($promotion, sprintf('There is no promotion %s in registry', $promotionName);
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingPromotionsContext.php [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    /**
+     * @Then /^(this promotion) should be coupon based$/
+     */
+    public function thisPromotionShouldBeCouponBased(PromotionInterface $promotion): void
+    {
+        $returnedPromotion = $this->getPromotionByPromotionNameFromResponseCollection(
+            $promotion->getName(),
+            $this->responseChecker->getCollection($this->apiAdminClient->getLastResponse())
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingPromotionsContext.php:79]: Maybe you could use here a `getCollectionItemsWithValue` method to filter out the promotions by name or by coupon based value

File: src/Sylius/Behat/Context/Api/Admin/ManagingPromotionsContext.php [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        );
+
+        Assert::true($returnedPromotion['couponBased']);
```
Comment (AdamKasp) [inline src/Sylius/Behat/Context/Api/Admin/ManagingPromotionsContext.php:82]:
```suggestion
        Assert::true($returnedPromotion['couponBased'], sprintf('The promotion %s isn\'t coupon based', $promotionName);
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingPromotionsContext.php [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    /**
+     * @Then /^I should be able to manage coupons for (this promotion)$/
+     */
+    public function iShouldBeAbleToManageCouponsForThisPromotion(PromotionInterface $promotion)
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingPromotionsContext.php:88]:
```suggestion
    public function iShouldBeAbleToManageCouponsForThisPromotion(PromotionInterface $promotion): void
```

File: src/Sylius/Behat/Resources/config/suites/api/promotion/managing_promotions.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+                - sylius.behat.context.setup.order
+                - sylius.behat.context.setup.payment
+                - sylius.behat.context.setup.product
+                - sylius.behat.context.setup.product_taxon
+                - sylius.behat.context.setup.promotion
+                - sylius.behat.context.setup.admin_api_security
+                - sylius.behat.context.setup.shipping
+                - sylius.behat.context.setup.taxonomy
+
+                - sylius.behat.context.api.admin.managing_promotions
```
Comment (GSadee) [inline src/Sylius/Behat/Resources/config/suites/api/promotion/managing_promotions.yml:34]: Do we need all of these contexts here (like e.g. for payment and shipping)?

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Promotion.xml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+                <attribute>admin:promotion:read</attribute>
+            </attribute>
+        </attribute>
+
+        <attribute name="validation_groups">sylius</attribute>
+
+        <collectionOperations>
+            <collectionOperation name="admin_get">
+                <attribute name="method">GET</attribute>
+                <attribute name="path">admin/promotions</attribute>
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Promotion.xml:30]: You could add `<attribute name="route_prefix">admin</attribute>` as all of the endpoints are for admin

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, @arti0090! :tada:

**Pull request merged**